lib/virt_feature_test_base.pm

```diff
@@ -48,9 +49,11 @@ sub run_test {
 sub prepare_run_test {
     my $self = shift;
 
-    script_run("rm -f /root/{commands_history,commands_failure}");
+    reconnect_when_ssh_console_broken unless defined(script_run("rm -f /root/{commands_history,commands_failure}", die_on_timeout => 0));
```

Comment:
what's the condition checking? Would you please explain a little bit?

Reply:
sometimes ssh console broke due to a command timeout, such as supportconfig,  in previous test, the current test will fail to start. The change is to reconnect ssh console to begin the test even if the ssh console is broken.

Reply:
yes, I get this part, but for the unless defined(script_run("rm -f /root/{commands_history,commands_failure}" part, why do you use this as condition?

Reply:
It was there. It is not added by me. I choose this command because it is the first ssh command in each test.

Reply:
It is the first command in each test because in some code, it would qeury /root/{commands_history,commands_failure}, which were introduced by Wayne. 

But here, you use it as condition -- unless defined(script_run("rm -f xx")), script_run returns 0 or 1, what does defined (0 or 1) mean? I do not understand. Are you using it to tell ssh connection status? If script_run returns 0,  call reconnect_when_ssh_console. Otherwise, not call it? If yes, then it is the same to remove define, I think.

Reply:
oh, you asked why 'defined' is used here. It is because I don't care the return value of script_run is '0' or '1',  either of them means the ssh console is connected. 

but if the ssh console broke, the return value is 'undefined', so here `unless defined(script_run())` is used.